crates/context/config/src/icp/types.rs

```diff
@@ -169,17 +161,19 @@ pub struct ICRequest {
     pub kind: ICRequestKind,
     pub signer_id: ICRepr<SignerId>,
     pub timestamp_ms: u64,
+    pub nonce: u64,
 }
 
 impl ICRequest {
-    pub fn new(signer_id: SignerId, kind: ICRequestKind) -> Self {
+    pub fn new(signer_id: SignerId, kind: ICRequestKind, nonce: u64) -> Self {
         Self {
             signer_id: ICRepr::new(signer_id),
             kind,
             timestamp_ms: SystemTime::now()
                 .duration_since(UNIX_EPOCH)
                 .expect("Time went backwards")
                 .as_millis() as u64,
```

Comment:
When are we removing this?

Reply:
Let me sync with Alen and will let you know